shared-bindings/bitmapfilter/__init__.c

```diff
@@ -0,0 +1,606 @@
+//|         gives the fraction of green to mix into green."""
+//|
+static const mp_obj_namedtuple_type_t bitmapfilter_channel_mixer_type = {
+    NAMEDTUPLE_TYPE_BASE_AND_SLOTS(MP_QSTR_ChannelMixer),
+    .n_fields = 9,
+    .fields = {
+        MP_QSTR_rr,
+        MP_QSTR_rg,
+        MP_QSTR_rb,
+        MP_QSTR_gr,
+        MP_QSTR_gg,
+        MP_QSTR_gb,
+        MP_QSTR_br,
+        MP_QSTR_bg,
+        MP_QSTR_bb,
+    },
+};
+
+//| class ChannelMixerOffset:
+//|     """A weight object to use with mix()"""
+//|
+//|     def __init__(
+//|         self,
+//|         rr: float,
+//|         rg: float,
+//|         rb: float,
+//|         r_add: float,
+//|         gr: float,
+//|         gg: float,
+//|         gb: float,
+//|         g_add: float,
+//|         br: float,
+//|         bg: float,
+//|         bb: float,
+//|         b_add: float,
+//|     ) -> None:
+//|         """Perform a mixing operation where each channel can receive a fraction of every other channel, plus an offset value.
+//|
+//|         The parameters with names like ``rb`` give the fraction of
+//|         each channel to mix into every other channel. For instance,
+//|         ``rb`` gives the fraction of blue to mix into red, and ``gg``
+//|         gives the fraction of green to mix into green. The ``_add``
+//|         parameters give an offset value to add to the channel."""
+//|
+static const mp_obj_namedtuple_type_t bitmapfilter_channel_mixer_offset_type = {
+    NAMEDTUPLE_TYPE_BASE_AND_SLOTS(MP_QSTR_ChannelMixerOffset),
+    .n_fields = 9,
+    .fields = {
+        MP_QSTR_rr,
+        MP_QSTR_rg,
+        MP_QSTR_rb,
+        MP_QSTR_r_add,
+        MP_QSTR_gr,
+        MP_QSTR_gg,
+        MP_QSTR_gb,
+        MP_QSTR_g_add,
+        MP_QSTR_br,
+        MP_QSTR_bg,
+        MP_QSTR_bb,
+        MP_QSTR_b_add,
+    },
+};
+
+//| def mix(
+//|     bitmap: displayio.Bitmap,
+//|     weights: ChannelScale | ChannelScaleOffset | ChannelMixer | ChannelMixerOffset,
+//|     mask: displayio.Bitmap | None = None,
+//| ) -> displayio.Bitmap:
+//|     """Perform a channel mixing operation on the bitmap
+//|
+//|     This is similar to the "channel mixer" tool in popular photo editing software.
+//|     Imagemagick calls this "-color-matrix". In PIL, this is accomplished with the
+//|     ``convert`` method's ``matrix`` argument.
+//|
+//|     The ``bitmap``, which must be in RGB565_SWAPPED format, is modified
+//|     according to the ``weights``.
+//|
+//|     If ``weights`` is a `ChannelScale`
+//|     object, then each channel is scaled independently: The
+//|     numbers are the red, green, and blue channel scales.
+//|
+//|     If ``weights`` is a `ChannelScaleOffset`
+//|     object, then each channel is scaled and offset independently:
+//|     The first two numbers are applied to the red channel: scale and
+//|     offset. The second two number are applied to the green channel,
+//|     and the last two numbers to the blue channel.
```

Comment:
Instead of explaining what number in each position does, it would be better to give examples when it is useful. The tuple has the individual number doc.